app/build.gradle

```diff
@@ -24,29 +24,29 @@ android {
         }
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
     }
     kotlinOptions {
-        jvmTarget = '1.8'
+        jvmTarget = JavaVersion.VERSION_17.toString()
```

Comment:
Could you explain why you are using Java 17?

Reply:
The JVM receives frequent updates and new features. Therefore, it's generally preferable for Kotlin to be able to utilize these new capabilities, rather than being constrained by the features available in Java 8.